dulwich/client.py

```diff
@@ -1557,10 +1557,9 @@ def __repr__(self):
 
     def _get_url(self, path):
         if not isinstance(path, str):
-            # TODO(jelmer): this is unrelated to the local filesystem;
-            # This is not necessarily the right encoding to decode the path
-            # with.
-            path = path.decode(sys.getfilesystemencoding())
+            # urllib3.util.url._encode_invalid_chars() converts the path back
+            # to bytes using the same parameters.
+            path = path.decode('utf-8', 'surrogatepass')
```

Comment:
Can we drop the second argument? Replacing characters seems like a recipe for confusing problems - it's better to just raise an error here and let the user pass in the right arguments.

Reply:
I dropped it.